adam-core/src/main/scala/org/bdgenomics/adam/util/FileExtensions.scala

```diff
@@ -0,0 +1,145 @@
+/**
+ * Licensed to Big Data Genomics (BDG) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The BDG licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.adam.util
+
+/**
+ * Match path names to file extensions.
+ */
+object FileExtensions {
+
+  /**
+   * @param pathName The path name to match.
+   * @return Returns true if the path name matches a 2bit format file extension.
+   */
+  private[adam] def is2BitExt(pathName: String): Boolean = {
+    pathName.endsWith(".2bit")
+  }
+
+  /**
+   * @param pathName The path name to match.
+   * @return Returns true if the path name matches a BAM/CRAM/SAM format file extension.
+   */
+  private[adam] def isBamExt(pathName: String): Boolean = {
+    pathName.endsWith(".bam") ||
+      pathName.endsWith(".cram") ||
+      pathName.endsWith(".sam")
+  }
+
+  /**
+   * @param pathName The path name to match.
+   * @return Returns true if the path name matches a BED6/12 format file extension.
+   */
+  private[adam] def isBedExt(pathName: String): Boolean = {
+    pathName.endsWith(".bed") ||
+      pathName.endsWith(".bed.gz") ||
```

Comment:
ok, I suppose we could do an isCompressed check in `ADAMContext`, since it has access to the Hadoop configuration, then trim off the extension before checking with `FileExtensions`